app/controllers/order_items_controller.rb

```diff
@@ -6,14 +5,25 @@ def create
     @order_item = @order.order_items.create(order_item_params)
     session[:order_id] = @order.id
 
-    redirect_to product_path(@order_item.product_id)
+    if order_item_params[:quantity].to_i > @order_item.product.stock.to_i
+      flash[:error] = "Unfortunately we don't have #{order_item_params[:quantity].to_i} #{@order_item.product.name}, only #{@order_item.product.stock.to_i} available"
+      redirect_to cart_path
+    else
+      redirect_to cart_path
```

Comment:
It seems like both methods will do this redirect, so the `else` should be removed and the redirect should be outside of the `if`